blackduck/HubRestApi.py

```diff
@@ -1240,6 +1241,32 @@ def get_project_info(self, project_name, link_name):
         else:
             return {} # nada
 
+    def get_project_violation_status(self, project_name, version):
+
+        project = self.get_project_by_name(project_name)
```

Comment:
did you consider using self.get_project_version_by_name? would replace use of get_project_by_name and the subsequent code to retrieve the version if present, e.g. this method would then become,

def get_project_violation_status(self, project_name, version):
    version = self.get_project_version_by_name(project_name, version)
    if version:
        return version.get('policyStatus', None)
    else:
        return "{}:{} not found".format(project_name, version)

or something along those lines?

Reply:
Ah, well I believe I was looking at it this way because it is a more precise error.   If I remember  get_project_version_by_name will return a non 200 on both missing projects or versions.  and I wanted to tell my users what was missing the project or the version or if the server was just miss-behaving